src/painting/painter.rs

```diff
@@ -208,6 +208,12 @@ impl Painter {
         use_ansi_coloring: bool,
         cursor_config: &Option<CursorConfig>,
     ) -> Result<()> {
+        // Reset any ANSI styling that may have been left by external commands
+        // This ensures the prompt is not affected by previous output styling
+        self.stdout
+            .queue(SetAttribute(Attribute::Reset))?
+            .queue(ResetColor)?;
```

Comment:
Are you sure `SetAttribute(Attribute::Reset)` isn't enough? Is `ResetColor` necessary?